util/src/main/java/io/kubernetes/client/util/Preconditions.java

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package io.kubernetes.client.util;
+
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+
+/**
```

Comment:
3. creating a dedicated utility for preconditions, to be re-used everywhere where `if(...) {throw new IllegalArgumentException...}`